src/libsyncengine/syncpal/syncpalworker.cpp

```diff
@@ -410,7 +412,8 @@ SyncStep SyncPalWorker::nextStep() const {
             logNbOps(ReplicaSide::Local);
             logNbOps(ReplicaSide::Remote);
 
-            if (!_syncPal->_computeFSOperationsWorker->getFileSizeMismatchMap().empty()) {
+            if (CommonUtility::isFileSizeMismatchDetectionEnabled() &&
+                !_syncPal->_computeFSOperationsWorker->getFileSizeMismatchMap().empty()) {
```

Comment:
An opportunity to create a public accessor `computeFSOperationsWorker()` for `SyncPal`.

Reply:
Done here: [d5d9f13](https://github.com/Infomaniak/desktop-kDrive/pull/269/commits/d5d9f1379e1d16d29d517eaece44cdb7cf9f3573)